src/nupic/encoders/BaseEncoder.hpp

```diff
@@ -0,0 +1,70 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2019, David McDougall
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ * ---------------------------------------------------------------------
+ */
+
+/** @file
+ * Defines the base class for all encoders.
+ */
+
+#ifndef NTA_ENCODERS_BASE
+#define NTA_ENCODERS_BASE
+
+#include <nupic/types/Sdr.hpp>
+
+namespace nupic {
```

Comment:
please use nupic::encoders to match with the rest of the codebase

Reply:
Ok, if encoders have their own namespace it will look like:
`nupic::encoders::ScalarEncoder( args )`